examples/metrics/projection.py

```diff
@@ -0,0 +1,382 @@
+#
+# Neither of these options is more correct than the other. Rather, they
+# represent different studies.
+#
+# Both projections can be performed with the :py:func:`p3.data.projection`
+# function, by passing different arguments.
+#
+# For the first projection, we now need to specify the names of two columns
+# ("size" and "precision") to define the problem:
+
+# sphinx_gallery_start_ignore
+df["precision"] = "FP32"
+combine_data = {
+    "size": ["128x128x128", "256x256x256"] * 4,
+    "precision": ["FP64"] * 8,
+    "implementation": (["Library 1"] * 2 + ["Library 2"] * 2) * 2,
+    "machine": ["Cluster 1"] * 4 + ["Cluster 2"] * 4,
+    "fom": [1.0, 4.0, 1.4, 4.2, 0.5, 2.0, 0.25, 1.0],
+}
+df = pd.concat([df, pd.DataFrame(combine_data)], ignore_index=True)
+# sphinx_gallery_end_ignore
+
+proj1 = p3.data.projection(
+    df,
+    problem=["size", "precision"],
+    application=["implementation"],
+    platform=["machine"],
+)
+print(proj1)
+
+# %%
+# The original "size" and "precision" columns have been removed, and their
+# values have been concatenated to form the new "problem" column.
+#
+# For the second projection, we just need to specify "size", exactly as we did
+# before:
+
+proj2 = p3.data.projection(
+    df,
+    problem=["size"],
+    application=["implementation"],
+    platform=["machine"],
+)
+print(proj2)
+
+# %%
+# This time, the original "size" column has been removed, but the "precision"
+# column remains.
+#
+# Clearly, the values provided to the projection function change the structure
+# of the resulting :py:class:`pandas.DataFrame`. But why does that matter?
+# Well, let's take a look at what happens if we compute the maximum "fom" for
+# each (problem, application, platform) tuple in our projected datasets:
+
+max1 = proj1.groupby(["problem", "application", "platform"])["fom"].max()
+print(max1)
+
+# %%
+
+max2 = proj2.groupby(["problem", "application", "platform"])["fom"].max()
+print(max2)
+
+# %%
+#
+# Similar :py:func:`pandas.DataFrame.groupby` calls form the backbone of many
+# functions provided by the P3 Analysis Library, since metrics like
+# "application efficiency" and "performance portability" ultimately depend on
+# an understanding of which variable combinations deliver the best performance.
+#
+# .. important::
+#     The selected projection can have significant impact on the results of
+#     subsequent analysis, and it is critical to ensure that the projection
+#     is correct before digging too deep into (or presenting!) any results.
+#
+# Next Steps
+# ----------
+#
+# After raw performance data has been projected onto definitions
+# of "problem", "application" and "platform", it can be passed to
+# any of the P3 Analysis Library functions that expect a
+# :py:class:`pandas.DataFrame`.
+#
+# The examples below show how to use projected data to compute
+# and visualize derived metrics like "application efficiency",
+# "performance portability" and "code divergence".
```

Comment:
+1 for the Oxford comma :) (and a couple of lines up too)

Reply:
I am not a big fan of Oxford commas, but I'll add them: 0cda9bd71b9f0f25f9657e22b12ef5be2d17ee26